src/mods/vr/runtimes/OpenXR.cpp

```diff
@@ -472,19 +486,72 @@ VRRuntime::Error OpenXR::update_matrices(float nearz, float farz) {
     std::unique_lock __{ this->eyes_mtx };
     std::unique_lock ___{ this->pose_mtx };
 
+    auto& vr = VR::get();
+
+    // TODO: check signs
+    this->raw_projections[0][0] = tan(this->views[0].fov.angleLeft);
+    this->raw_projections[0][1] = tan(this->views[0].fov.angleRight);
+    this->raw_projections[0][2] = tan(this->views[0].fov.angleUp);
+    this->raw_projections[0][3] = tan(this->views[0].fov.angleDown);
+    this->raw_projections[1][0] = tan(this->views[1].fov.angleLeft);
+    this->raw_projections[1][1] = tan(this->views[1].fov.angleRight);
+    this->raw_projections[1][2] = tan(this->views[1].fov.angleUp);
+    this->raw_projections[1][3] = tan(this->views[1].fov.angleDown);
+    // SPDLOG_INFO("Original left  {}, {}, {}, {}", this->raw_projections[0][0], this->raw_projections[0][1],this->raw_projections[0][2], this->raw_projections[0][3]);
+    // SPDLOG_INFO("Original right {}, {}, {}, {}", this->raw_projections[1][0], this->raw_projections[1][1],this->raw_projections[1][2], this->raw_projections[1][3]);
+
     for (auto i = 0; i < 2; ++i) {
         const auto& pose = this->views[i].pose;
         const auto& fov = this->views[i].fov;
 
         // Update projection matrix
         //XrMatrix4x4f_CreateProjection((XrMatrix4x4f*)&this->projections[i], GRAPHICS_D3D, tan(fov.angleLeft), tan(fov.angleRight), tan(fov.angleUp), tan(fov.angleDown), nearz, farz);
 
+        // NOTE the sign convention for left-right is opposite to how it is in OpenVR. Up/down is the same
         auto get_mat = [&](int eye) {
-            const auto top = tan(fov.angleUp);
-            const auto bottom = tan(fov.angleDown);
-            const auto left = tan(fov.angleLeft);
-            const auto right = tan(fov.angleRight);
+            const auto tan_half_fov = new float[4];
```

Comment:
Very big alarm bells when I saw this. Do NOT call `new`, we do not allocate raw pointers in modern C++ without intention, we prefer `std::unique_ptr` or `std::shared_ptr`. We also do not call `new` without calling `delete`. This is creating a memory leak every time this is called.

Change this to `std::array<float, 4> tan_half_fov{};`.